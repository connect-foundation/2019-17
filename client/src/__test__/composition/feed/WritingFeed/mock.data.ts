export const content = '테스트 피드쓰기';
