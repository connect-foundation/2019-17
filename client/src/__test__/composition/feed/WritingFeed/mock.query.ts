import { ENROLL_FEED_MUTATION } from 'composition/Feed/WritingFeed/WritingFeed.query';
import { content } from './mock.data';
import { ENROLL_WRITING_FEED } from 'cache/writingFeed.query';

export const mocks = [
  {
    request: {
      query: ENROLL_FEED_MUTATION,
      variables: {
        content,
        files: []
      }
    },
    result: {
      data: {
        enrollFeed: true
      }
    }
  },
  {
    request: {
      query: ENROLL_FEED_MUTATION,
      variables: {
        content: '',
        files: []
      }
    },
    result: {
      data: {
        enrollFeed: false
      }
    }
  },
  {
    request: {
      query: ENROLL_WRITING_FEED,
      variables: {
        content: ''
      }
    },
    result: {
      data: {
        enrollWritingFeed: null
      }
    }
  },
  {
    request: {
      query: ENROLL_WRITING_FEED,
      variables: {
        content
      }
    },
    result: {
      data: {
        enrollWritingFeed: null
      }
    }
  }
];
