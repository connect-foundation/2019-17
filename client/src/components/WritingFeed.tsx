import React, { useState } from 'react';
import styled from 'styled-components';
import { MdPhotoSizeSelectActual } from 'react-icons/md';
import TextareaAutosize from 'react-textarea-autosize';
import Button from './Button';

const Form = styled.form`
  width: 500px;
  min-height: 150px;
  border: 1px solid ${props => props.theme.colors.borderColor};
`;

const Header = styled.div`
  height: 32px;
  background-color: #f5f6f7;
  border-bottom: 1px solid ${props => props.theme.colors.borderColor};
  color: ${props => props.theme.colors.textColor};
  display: flex;
  align-items: center;
  padding: 0.5rem;
  font-size: 0.8rem;
  font-weight: 600;
`;

const Footer = styled.div`
  height: 50px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-left: 1rem;
  padding-right: 1rem;
  color: ${props => props.theme.colors.textColor};
  background-color: white;
`;

const ContentContainer = styled.div`
  display: flex;
  width: 100%;
  border-bottom: 1px solid ${props => props.theme.colors.borderColor};
`;

const ContentWrapper = styled.div`
  padding-left: 1rem;
  padding-right: 1rem;
`;

const Content = styled(TextareaAutosize)`
  all: unset;
  width: 100%;
  box-sizing: border-box;
  padding: 0.5rem;
  padding-top: 30px;
  font-size: 1.1rem;
  color: #1d2028;
  min-height: 70px;
  &::placeholder {
    font-size: 0.8rem;
  }
`;

const ProfileColumn = styled.div`
  width: 65px;
  height: 100%;
`;

const UploadInput = styled.input`
  position: absolute;
  top: 0px;
  left: 0px;
  opacity: 0;
`;

const UploadButton = styled.label`
  all: unset;
  background-color: #f5f6f7;
  width: 110px;
  height: 35px;
  border-radius: 10px;
  display: flex;
  align-items: center;
  padding-left: 0.5rem;
  cursor: pointer;
  transition: box-shadow 0.3s ease-in-out;
  &:hover {
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  }
`;

const PhotoText = styled.span`
  line-height: 35px;
  font-size: 0.9rem;
  font-weight: 500;
`;

const PhotoIcon = styled(MdPhotoSizeSelectActual)`
  color: ${props => props.theme.colors.facebookBlue};
  font-size: 1.5rem;
  margin-right: 0.3rem;
`;

function WritingFeed() {
  const [content, setContent] = useState('');
  const onChange = (e: React.ChangeEvent<HTMLTextAreaElement>): void => {
    setContent(e.target.value);
  };
  return (
    <Form encType="multipart/form-data">
      <Header>게시물 만들기</Header>
      <ContentWrapper>
        <ContentContainer>
          <ProfileColumn></ProfileColumn>
          <Content
            onChange={onChange}
            placeholder={'게시물 작성'}
            value={content}
          />
        </ContentContainer>
      </ContentWrapper>
      <UploadInput id="upload" type="file" />
      <Footer>
        <UploadButton htmlFor="upload">
          <PhotoIcon />
          <PhotoText>사진 업로드</PhotoText>
        </UploadButton>
        <Button text={'게시'} />
      </Footer>
    </Form>
  );
}

export default WritingFeed;
