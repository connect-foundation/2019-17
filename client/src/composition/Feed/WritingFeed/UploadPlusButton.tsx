import React from 'react';
import styled from 'styled-components';
import { lighten } from 'polished';
import { FaPlus } from 'react-icons/fa';

const ButtonContainer = styled.label`
  border: 3px dashed ${props => props.theme.colors.borderColor};
  border-radius: ${props => props.theme.borders.radius};
  width: 100px;
  height: 100px;
  color: ${props => props.theme.colors.borderColor};
  display: flex;
  justify-content: center;
  align-items: center;
  transition: color 0.3s linear;
  &:hover {
    color: ${props => lighten(0.2, props.theme.colors.textColor)};
  }
  cursor: pointer;
`;

function UploadPlusButton({ targetId }: { targetId: string }) {
  return (
    <ButtonContainer htmlFor={targetId}>
      <FaPlus />
    </ButtonContainer>
  );
}

export default UploadPlusButton;
