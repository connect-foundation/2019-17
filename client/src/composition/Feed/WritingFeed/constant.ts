export const PLACEHOEDER_TEXT = '게시물 작성';
