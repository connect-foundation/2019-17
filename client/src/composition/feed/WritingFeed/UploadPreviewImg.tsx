import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  position: relative;
`;

const Image = styled.img`
  position: absolute;
  top: 0;
  left: 0;
  width: 100px;
  height: 100px;
  border-radius: ${props => props.theme.borders.radius};
`;

const Overlay = styled.div`
  opacity: 0;
  color: rgba(0, 0, 0, 0.7);
  &:hover {
    opacity: 1;
  }
`;

interface IProps {
  imageUrl: string;
}

function UploadPreviewImg({ imageUrl }: IProps) {
  return (
    <Container>
      <Image src={imageUrl} />
      <Overlay />
    </Container>
  );
}

export default UploadPreviewImg;
