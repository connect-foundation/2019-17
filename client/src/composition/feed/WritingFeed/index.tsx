import WritingFeedContainer from './WritingFeedContainer';
export default WritingFeedContainer;
