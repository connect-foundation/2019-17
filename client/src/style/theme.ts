import { DefaultTheme, css } from 'styled-components';

const theme: DefaultTheme = {
  colors: {
    facebookBlue: '#4267B2',
    bgColor: '#EAEBEE',
    white: '#FFFFFF',
    facebookTextColor: '#4B4F56',
    fontMainBlue: '#385898',
    fontButtonGray: '#606770',
    borderColor: '#dddfe2',
    textColor: '#4b4f56'
  },
  borders: {
    borderStyle: '1px solid #ccd0d5',
    radius: '5px',
    feedBorder: css`
      border: 1px solid #dddfe2;
      border-radius: 3px;
    `
  },
  alarmWidth: '430px',
  alarmHeight: '550px'
};

export default theme;
