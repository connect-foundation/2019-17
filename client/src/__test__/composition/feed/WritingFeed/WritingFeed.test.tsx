import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import WritingFeed from 'composition/Feed/WritingFeed';
import { ThemeProvider } from 'styled-components';
import theme from 'style/theme';
import { ApolloProvider } from '@apollo/react-hoc';
import client from 'apollo/ApolloClient';

afterEach(cleanup);

test('<WritingFeed />', () => {
  const { container } = render(
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <WritingFeed />
      </ThemeProvider>
    </ApolloProvider>
  );
  expect(container).toMatchSnapshot();
});
