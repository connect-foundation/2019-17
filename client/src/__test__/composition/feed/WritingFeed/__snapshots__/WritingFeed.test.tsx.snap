// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WritingFeed /> 1`] = `
<div>
  <form
    class="sc-gzVnrw hxyAtP"
    enctype="multipart/form-data"
  >
    <div
      class="sc-htoDjs eXXuCg"
    >
      게시물 만들기
    </div>
    <div
      class="sc-gZMcBi eCVzxD"
    >
      <div
        class="sc-iwsKbI dYPrPu"
      >
        <div
          class="sc-VigVT jDRjwl"
        >
          <img
            alt="profile image"
            class="sc-bwzfXH hNFusQ"
            src="/images/profile.jpg"
          />
        </div>
        <textarea
          class="sc-gqjmRU ficjwu"
          placeholder="게시물 작성"
          style="height: 2px;"
        />
      </div>
      <div
        class="sc-iwsKbI dYPrPu"
      />
    </div>
    <input
      accept="image/*"
      class="sc-jTzLTM eFoXsh"
      id="upload"
      type="file"
    />
    <div
      class="sc-dnqmqq hitCSE"
    >
      <label
        class="sc-fjdhpX cvPITL"
        for="upload"
      >
        <svg
          class="sc-cSHVUG jgwGnI"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 3H3C2 3 1 4 1 5v14c0 1.1.9 2 2 2h18c1 0 2-1 2-2V5c0-1-1-2-2-2zM5 17l3.5-4.5 2.5 3.01L14.5 11l4.5 6H5z"
          />
        </svg>
        <span
          class="sc-jzJRlG iWvITD"
        >
          사진 업로드
        </span>
      </label>
      <button
        class="sc-bdVaJa lnkbBL"
      >
        게시
      </button>
    </div>
  </form>
</div>
`;
